apiserver/facades/controller/crossmodelsecrets/crossmodelsecrets.go

```diff
@@ -141,7 +141,7 @@ func (s *CrossModelSecretsAPI) checkRelationMacaroons(ctx stdcontext.Context, co
 	// A cross model secret can only be accessed if the corresponding cross model relation
 	// it is scoped to is accessible by the supplied macaroon.
 	auth := s.authCtxt.Authenticator()
-	return auth.CheckRelationMacaroons(ctx, s.modelUUID, offerUUID, names.NewRelationTag(relKey), mac, version)
+	return auth.CheckRelationMacaroons(ctx, model.UUID(s.modelUUID), offerUUID, names.NewRelationTag(relKey), mac, version)
```

Comment:
This value is still coming from state in the register function. Can we please swap this over to a service call and update the type on to core and also the name ?

Reply:
all done